presto-main/src/main/java/io/prestosql/util/CIDR.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.util;
+
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.StringTokenizer;
+
+public abstract class CIDR
```

Comment:
All of this seems overkill (CIDR, CIDR4, CIDR6 classes and abstractions). We should be able to implement the `contains` function like this (it might need a few adjustments, but it serves as a starting point):

```java
int separator = cidr.indexOf("/");
if (separator == -1) {
    throw new IllegalArgumentException("Invalid CIDR");
}

int prefixLength = Integer.parseInt(cidr.substring(separator + 1));

if (prefixLength < 0) {
    throw new IllegalArgumentException("Invalid prefix length");
}

if (prefixLength == 0) {
    return true;
}

byte[] base = InetAddresses.forString(cidr.substring(0, separator)).getAddress();
int baseLength = base.length * Byte.SIZE;

if (prefixLength > baseLength) {
    throw new IllegalArgumentException("Prefix length exceeds address length");
}

BigInteger mask = BigInteger.ONE
        .shiftLeft(prefixLength + 1)
        .subtract(BigInteger.ONE)
        .shiftLeft(baseLength - prefixLength);

BigInteger maskedBase = new BigInteger(base)
        .and(mask);

BigInteger maskedAddress = new BigInteger(InetAddresses.forString(ip).getAddress())
        .and(mask);

return maskedBase.equals(maskedAddress);
```

Reply:
In my understanding, we need to write redundant code to cast ipaddress to string. I changed `InetAddresses.forString(ip)` to `InetAddress.getByAddress(address)` using byte[] type.

```java
utf8Slice(InetAddresses.toAddrString(InetAddress.getByAddress(address.getBytes()))).toStringUtf8()
```

Reply:
InetAddress.getByAddress will do a DNS lookup if the address is not an option address, so we don’t want that.

Reply:
InetAddresses.forString uses [InetAddress.getByAddress](https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#getByAddress-byte:A-) internally and the document says *no reverse name service lookup is performed*, so I think the change is safe.  Am I missing something? 

```java
  public static InetAddress forString(String ipString) {
    byte[] addr = ipStringToBytes(ipString);

    // The argument was malformed, i.e. not an IP string literal.
    if (addr == null) {
      throw formatIllegalArgumentException("'%s' is not an IP string literal.", ipString);
    }

    return bytesToInetAddress(addr);
  }

  private static InetAddress bytesToInetAddress(byte[] addr) {
    try {
      return InetAddress.getByAddress(addr);
    } catch (UnknownHostException e) {
      throw new AssertionError(e);
    }
  }
```

Reply:
Sorry, you’re right. I got it mixed up with getByName